### ISSUE TITLE
"Unsafe dynamic component" build errors when Embroider is enabled

### ISSUE THREAD

**Issue by roomman** [CrowdStrike/ember-headless-form#131]
Hi, when updating to the latest release (beta-2) I'm seeing the following error when building an Embroider-enabled application:
```
$TMPDIR/embroider/a423b3/node_modules/ember-headless-form/dist/components/headless-form.js/headless-form.js: Unsafe dynamic component: this.FieldComponent in node_modules/ember-headless-form/dist/components/headless-form.js
```
Would it be helpful if I provided a reproduction?

[From this](https://github.com/embroider-build/embroider/blob/main/docs/replacing-component-helper.md), it sounds like `ensure-safe-component` helper from the `@embroider/util` needs to be included wherever you've used the `(component)` helper to allow for static analysis.

**Comment by simonihmig** [comment]
Oh, I think I know what's going on. It's basically a bug in Embroider, that has been fixed already (https://github.com/embroider-build/embroider/pull/1383), but it's not released yet in a stable version. 

There are "unstable" releases (see https://www.npmjs.com/package/@embroider/core?activeTab=versions) you _could_ use. That's what the addon is also doing to get its Embroider tests to pass, see [here](https://github.com/CrowdStrike/ember-headless-form/blob/main/test-app/config/ember-try.js#L58-L60). Or you would need to wait for the stable release, which I know folks are actively working on, but it could take a couple more days...

**Comment by simonihmig** [comment]
The new version of this addon uses the new `<template>` tag (aka [template-imports](https://github.com/ember-template-imports/ember-template-imports)), which makes the template operate in so called strict mode. But strict mode does not allow any string-based lookups anyway (which normally makes Embroider fail when it cannot determine what you are trying to lookup with patterns like `{{component this.someThingEmbroiderCannotUnderstand}}`), but only allows explicit imports. So in that case Embroider can safely assume anything that is used in a template is already imported elsewhere, so it doesn't have to fail on `{{component this.someThingEmbroiderCannotUnderstand}}` anymore.

tl;dr in strict mode you never need `ensure-safe-component`.

The PR I linked above is fixing the bug that `<template>` was not correctly making Embroider go into strict mode, so that's why this fix is needed.

**Comment by roomman** [comment]
Excellent, thanks @simonihmig. Great to have clarified the relationship between "strict mode" and `<template>`. It's filled in plenty of gaps in my reading of other issues and RFCs. 🙏🏼
